GenericRelatedField raise It is redundant to specify source on field

**Background:**
I am using `django-activity-stream`. One instance of `Action` record contains multiple `GenericForeignKey`
My serializer have to represent `actor, action_object, and target` which are `GenericForeignKey`

**Problem:**
When I add more than 1 `GenericRelatedField` to the `serializer` it raise 
```
AssertionError: It is redundant to specify `source=''` on field 'TimelineUserSerializer' in serializer 'GenericRelatedField', because it is the same as the field name. Remove the `source` keyword argument.
```

Actually. I haven't put any `source` keyword argument in my `class`
```python
class TimelineUserSerializer(serializers.ModelSerializer):
    """For timeline keep it short"""
    class Meta:
        model = User
        fields = [
            'id',
            'username',
            'name',
        ]
```

`serializers.py`
```python
class TimelineSerializer(serializers.ModelSerializer):
    """Design for `read-only`"""
    __basic_fields = {
        User: TimelineUserSerializer(),
        Tweet: ShortTweetSerializer(),
        ReTweet: ReTweetSerializer(),
        Reply: ShortReplySerializer(),
        Picture: PictureSerializer(),
        Like: LikeSerializer(),
    }
    actor = GenericRelatedField(__basic_fields)
    action_object = GenericRelatedField(__basic_fields)
    target = GenericRelatedField(__basic_fields)

    class Meta:
        model = Action
        fields = [
            'id',
            'actor',
            'verb',
            'description',
            'target',
            'action_object',
            'timestamp',
            'public',
        ]
```

`python 3.7`
`requirements.txt`
```
-e git+git@github.com:justquick/django-activity-stream.git@master#egg=django-activity-stream
-e git+git@github.com:sunscrapers/djoser.git@master#egg=djoser
alabaster==0.7.11         # via sphinx
amqp==2.3.2               # via kombu
appnope==0.1.0            # via ipython
argon2-cffi==18.1.0
arrow==0.12.1
atomicwrites==1.1.5       # via pytest
attrs==18.1.0             # via pytest
babel==2.6.0              # via flower, sphinx
backcall==0.1.0           # via ipython
billiard==3.5.0.4         # via celery
boto3==1.7.62             # via django-storages
botocore==1.10.62         # via boto3, s3transfer
celery==4.2.1
certifi==2018.4.16        # via requests
cffi==1.11.5              # via argon2-cffi
chardet==3.0.4            # via requests
collectfast==0.6.2
coreapi==2.3.3
coreschema==0.0.4         # via coreapi
coverage==4.5.1
decorator==4.3.0          # via ipython, traitlets
defusedxml==0.5.0         # via python3-openid
django-allauth==0.36.0
django-anymail[mailgun]==3.0
django-celery-results==1.0.1
django-cors-headers==2.4.0
django-coverage-plugin==1.5.0
django-crispy-forms==1.7.2
django-debug-toolbar==1.9.1
django-environ==0.4.5
django-extensions==2.1.0
django-filter==2.0.0
django-model-utils==3.1.2
django-redis==4.9.0
django-storages[boto3]==1.6.6
django-templated-mail==1.1.1
django==2.0.7
djangorestframework-jwt==1.11.0
djangorestframework==3.8.2
docutils==0.14            # via botocore, sphinx
factory-boy==2.11.1
faker==0.8.17             # via factory-boy
flake8==3.5.0
flower==0.9.2
gunicorn==19.9.0
idna==2.7                 # via requests
imagesize==1.0.0          # via sphinx
ipdb==0.11
ipython-genutils==0.2.0   # via traitlets
ipython==6.4.0            # via ipdb
itypes==1.1.0             # via coreapi
jedi==0.12.1              # via ipython
jinja2==2.10              # via coreschema, sphinx
jmespath==0.9.3           # via boto3, botocore
kombu==4.2.1              # via celery
markupsafe==1.0           # via jinja2
mccabe==0.6.1             # via flake8
model-mommy==1.6.0
more-itertools==4.2.0     # via pytest
oauthlib==2.1.0           # via requests-oauthlib
packaging==17.1           # via sphinx
parso==0.3.1              # via jedi
pexpect==4.6.0            # via ipython
pickleshare==0.7.4        # via ipython
pillow==5.2.0
pluggy==0.6.0             # via pytest
prompt-toolkit==1.0.15    # via ipython
psycopg2-binary==2.7.5
psycopg2==2.7.5
ptyprocess==0.6.0         # via pexpect
py==1.5.4                 # via pytest
pycodestyle==2.3.1        # via flake8
pycparser==2.18           # via cffi
pyflakes==1.6.0           # via flake8
pygments==2.2.0           # via ipython, sphinx
pyjwt==1.6.4              # via djangorestframework-jwt
pyparsing==2.2.0          # via packaging
pytest-django==3.3.2
pytest-sugar==0.9.1
pytest==3.6.3
python-dateutil==2.7.3    # via arrow, botocore, faker
python-slugify==1.2.5
python3-openid==3.1.0     # via django-allauth
pytz==2018.5
raven==6.9.0
redis==2.10.6
requests-oauthlib==1.0.0  # via django-allauth
requests==2.19.1          # via coreapi, django-allauth, django-anymail, requests-oauthlib, sphinx
rest-framework-generic-relations==1.2.0
s3transfer==0.1.13        # via boto3
simplegeneric==0.8.1      # via ipython
six==1.11.0               # via argon2-cffi, django-anymail, django-coverage-plugin, django-extensions, faker, model-mommy, more-itertools, packaging, prompt-toolkit, pytest, python-dateutil, sphinx, traitlets
snowballstemmer==1.2.1    # via sphinx
sphinx==1.7.6
sphinxcontrib-websupport==1.1.0  # via sphinx
sqlparse==0.2.4           # via django-debug-toolbar
termcolor==1.1.0          # via pytest-sugar
text-unidecode==1.2       # via faker
tornado==5.1              # via flower
traitlets==4.3.2          # via ipython
unidecode==1.0.22         # via python-slugify
uritemplate==3.0.0        # via coreapi
urllib3==1.23             # via requests
vine==1.1.4               # via amqp
wcwidth==0.1.7            # via prompt-toolkit
werkzeug==0.14.1
```

Is it bug?

--- a/Changelog.md
+++ b/Changelog.md
@@ -1,5 +1,8 @@
 # Rest Framework Generic Relations Changelog
 
+## v1.2.1
+* Add error handling for reusing a `Serializer` instance in a `GenericRelatedField`.
+
 ## v1.2.0
 
 * Add Django 2.0 support (need Python min 3.4, Django min. 1.11).

--- a/generic_relations/serializers.py
+++ b/generic_relations/serializers.py
@@ -29,6 +29,9 @@ class GenericSerializerMixin(object):
         super(GenericSerializerMixin, self).__init__(*args, **kwargs)
         self.serializers = serializers
         for serializer in self.serializers.values():
+            if serializer.source is not None:
+                msg = '{}() cannot be re-used. Create a new instance.'
+                raise RuntimeError(msg.format(type(serializer).__name__))
             serializer.bind('', self)
 
     def to_internal_value(self, data):

--- a/generic_relations/tests/test_relations.py
+++ b/generic_relations/tests/test_relations.py
@@ -463,3 +463,18 @@ class TestGenericRelatedFieldDeserialization(TestCase):
 
             self.assertEqual(len(w), 1)
             self.assertIs(w[0].category, DeprecationWarning)
+
+
+class TestGenericRelatedField(TestCase):
+    def test_multiple_declaration(self):
+        with self.assertRaises(RuntimeError):
+            class TagSerializer(serializers.ModelSerializer):
+                fields = {
+                    Bookmark: BookmarkSerializer(),
+                    Note: NoteSerializer(),
+                }
+                first = GenericRelatedField(fields)
+                second = GenericRelatedField(fields)
+
+                class Meta:
+                    model = Tag

